internal/security/secretstore/passwordprovider.go

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package secretstore
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"syscall"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+type PasswordProvider struct {
+	loggingClient        logger.LoggingClient
+	execRunner           ExecRunner
+	initialized          bool
+	passwordProvider     string
+	passwordProviderArgs []string
+	resolvedPath         string
+}
+
+// NewPasswordProvider creates a new PasswordProvider
+func NewPasswordProvider(lc logger.LoggingClient, execRunner ExecRunner) *PasswordProvider {
+	return &PasswordProvider{
+		loggingClient: lc,
+		execRunner:    execRunner,
+	}
+}
+
+// SetConfiguration parses token provider configuration and resolves paths specified therein
+func (p *PasswordProvider) SetConfiguration(passwordProvider string, passwordProviderArgs []string) error {
+	var err error
+	p.passwordProvider = passwordProvider
+	p.passwordProviderArgs = passwordProviderArgs
```

Comment:
Should there be any validation of these strings (i.e. max length)?

Reply:
PasswordProvider is validated by LookPath.  ProviderArgs there is no obvious validation criteria.  Also, since this is the password generation function, the input is implicitly trusted.  (If the input can be forged, it can run pretty much anything.)

Reply:
OK, if you're OK with it, then I'll trust your judgement on this. The only reason I brought this up is because of the recent GRub2 BootHole vulnerability which was involves buffer overruns involving the Grub2 config file...